test/cql-pytest/test_tablets.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2023-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+#############################################################################
+# Some tests for the new "tablets"-based replication, replicating the old
+# "vnodes". Eventually, ScyllaDB will use tablets by default and all tests
+# will run using tablets, but these tests are for specific issues discovered
+# while developing tablets that didn't exist for vnodes. Note that most tests
+# for tablets require multiple nodes, and are in the test/topology*
+# directory, so here we'll probably only ever have a handful of single-node
+# tests.
+#############################################################################
+
+import pytest
+from util import unique_name
+from cassandra.protocol import ConfigurationException
+
+# A fixture similar to "test_keyspace", just creates a keyspace that enables
+# tablets with initial_tablets=128
+# The "initial_tablets" feature doesn't work if the "tablets" experimental
+# feature is not turned on; In such a case, the tests using this fixture
+# will be skipped.
+@pytest.fixture(scope="module")
+def test_keyspace_128_tablets(cql, this_dc):
+    name = unique_name()
+    try:
+        cql.execute("CREATE KEYSPACE " + name + " WITH REPLICATION = { 'class' : 'NetworkTopologyStrategy', '" + this_dc + "': 1, 'initial_tablets': 128 }")
+    except ConfigurationException:
+        pytest.skip('Scylla does not support initial_tablets, or the tablets feature is not enabled')
+    yield name
+    cql.execute("DROP KEYSPACE " + name)
+
+# In the past (issue #16493), repeatedly creating and deleting a table
+# would leak memory. Creating a table with 128 tablets would make this
```

Comment:
We don't actually leak any memory. The memory is accounted for and eventually freed, this is true for both the task objects themselves and any memory they might pin. The problem this test is reproducing is a problem of waste. A retired task object held on to memory it no longer needed. The task object is eventually destroyed and this memory is freed. But in the meanwhile this memory is not available, leading to an OOM after a certain amount of task objects are accumulated. The seastar fix just reduces this waste memory significantly. There is no leak fixed,because there is no leak in the first place.

Reply:
If unused memory is kept forever - or even if "forever" is really "only" for 10 minutes (I don't know what is that TTL you speak of) then it's known as a memory leak.

Reply:
Looks like we have different definitions for "leak". It is not worth arguing over which is the correct one here.

Reply:
Whatever is the correct definition, if "something" (leak or not) allows a loop to deplete all memory, we have to fix it :-)